pkg/querier/worker/scheduler_processor.go

```diff
@@ -89,7 +90,7 @@ func (sp *schedulerProcessor) processQueriesOnSingleStream(workerCtx context.Con
 	// Run the querier loop (and so all the queries) in a dedicated context that we call the "execution context".
 	// The execution context is cancelled once the workerCtx is cancelled AND there's no inflight query executing.
 	execCtx, execCancel, inflightQuery := newExecutionContext(workerCtx, sp.log)
-	defer execCancel()
+	defer execCancel(errors.New("scheduler processor execution context canceled"))
```

Comment:
should we include more context here? the comment above makes it pretty clear when we would be cancelling the execution context
```
// The execution context is cancelled once the workerCtx is cancelled AND there's no inflight query executing.
```

Reply:
> should we include more context here? the comment above makes it pretty clear when we would be cancelling the execution context

yep it is pretty much what is in the comment, but the comment won't be present on the log line so you won't have access to it when querying the logs.